cli/azd/pkg/tools/git/git.go

```diff
@@ -216,6 +212,26 @@ func (cli *gitCli) PushUpstream(ctx context.Context, repositoryPath string, orig
 	return nil
 }
 
+func (cli *gitCli) ListStagedFiles(ctx context.Context, repositoryPath string) (string, error) {
+	runArgs := exec.NewRunArgs("git", "-C", repositoryPath, "ls-files", "--stage")
+	res, err := cli.commandRunner.Run(ctx, runArgs)
+	if err != nil {
+		return "", fmt.Errorf("failed to list files: %s: %w", res.String(), err)
+	}
+
+	return res.Stdout, nil
+}
```

Comment:
I would consider modifying this method to return an array of some structure that has already parsed the string output into the various parts so that other consumers of this API don't need to do the manual parsing.  

This git command was new to me but looks like it lists the files in the repo including the file attrs, some id/hash and the file path.

If you split this out into the struct then the filter of exec files really just comes down to filtering down the the array that match the right file attrs.

Reply:
I'm going to defer this post this PR. I can't see how we might need this for other places yet, but if we need to add it, we can easily refactor this later.